src/overlays/actors/ovl_En_Osn/z_en_osn.c

```diff
@@ -74,59 +101,824 @@ static DamageTable D_80AD2550 = {
+                (gSaveContext.entranceIndex == 0xC060)) {
+                this->unk_1EA |= 1;
+            }
+            this->unk_1F0 = 1;
+            if (globalCtx->sceneNum == SCENE_INSIDETOWER) {
+                if ((gSaveContext.entranceIndex == 0xC020) || (gSaveContext.entranceIndex == 0xC060)) {
+                    this->actionFunc = func_80AD16A8;
+                    return;
+                }
+                if (gSaveContext.entranceIndex == 0xC030) {
+                    func_80AD1398(this);
+                    this->actionFunc = func_80AD1634;
+                    return;
+                }
+                func_80AD144C(this, globalCtx);
+                return;
+            }
+            func_80AD144C(this, globalCtx);
+            break;
+
+        case 1:
+            this->unk_1EC = 0xF;
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            break;
+
+        case 2:
+            this->unk_1EC = 0x10;
+            Actor_ChangeAnimation(&this->skelAnime, sAnimations, this->unk_1EC);
+            this->actionFunc = EnOsn_Idle;
+            break;
+
+        case 3:
+            this->actor.flags &= ~ACTOR_FLAG_1;
+            this->actionFunc = func_80AD16A8;
+            break;
+
+        default:
+            Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void EnOsn_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOsn* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnOsn_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnOsn* this = THIS;
+    u32 sp34;
+
+    sp34 = Flags_GetSwitch(globalCtx, 0);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveWithGravity(&this->actor);
+    SkelAnime_Update(&this->skelAnime);
+    if (!(ENOSN_GET_3(&this->actor))) {
+        if (sp34 != 0) {
+            this->actor.flags |= ACTOR_FLAG_1;
+            func_80AD0830(this, globalCtx);
+            this->actor.draw = EnOsn_Draw;
+        } else {
+            this->actor.draw = NULL;
+            this->actor.flags &= ~ACTOR_FLAG_1;
+        }
+    }
+    func_800E9250(globalCtx, &this->actor, &this->unk_1D8, &this->unk_1DE, this->actor.focus.pos);
+}
+
+s32 EnOsn_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
+                           Gfx** gfx) {
+    EnOsn* this = (EnOsn*)thisx;
+
+    if (this->unk_1F0 && limbIndex == 11) {
+        Matrix_InsertXRotation_s(this->unk_1D8.y, MTXMODE_APPLY);
+    }
+    if ((this->unk_1EC == 9 || this->unk_1EC == 8) && limbIndex == 10) {
+        *dList = NULL;
+    }
+    return false;
+}
+
+void EnOsn_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor, Gfx** gfx) {
+    EnOsn* this = (EnOsn*)actor;
```

Comment:
```suggestion
void EnOsn_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
    EnOsn* this = (EnOsn*)thisx;
```
Looks like this didn't get changed properly.